test/Common.cpp

```diff
@@ -207,6 +212,41 @@ bool CommonOptions::parse(int argc, char const* const* argv)
 	return true;
 }
 
+string CommonOptions::toString(vector<string> const& _selectedOptions) const
+{
+	if (!m_singleton || _selectedOptions.empty())
+		return "";
```

Comment:
This function operates on the current instance of options and does not access `m_singleton`. Is this condition actually needed?

The ` _selectedOptions.empty()` part is fine but it seems that function would still work fine without it so I'd remove it. It's such a useless case that it's not really worth optimizing for.

Reply:
Yes, I should have changed that once I made it become a member of the struct.